src/lib/features/feature-toggle/feature-toggle-service.ts

```diff
@@ -387,6 +386,22 @@ class FeatureToggleService {
         }
     }
 
+    validateContraintValuesLimit(updatedConstrains: IConstraint[]) {
+        if (!this.flagResolver.isEnabled('resourceLimits')) return;
+
+        const limit = this.resourceLimits.constraintValues;
+        const constraintOverLimit = updatedConstrains.find(
+            (constraint) =>
+                Array.isArray(constraint.values) &&
+                constraint.values?.length > limit,
+        );
+        if (constraintOverLimit) {
+            throw new BadDataError(
+                `Constraint values limit of ${limit} is exceeded for ${constraintOverLimit.contextName}.`,
+            );
```

Comment:
I think we should use the Exceeds Limit error (`src/lib/error/exceeds-limit-error.ts`) for this instead of redoing the same thing ourselves. 

What do you think?

Reply:
Oh, didn't know we have it. I will adjust it in another PR. I think we can err on the side of consistency over precise contextual message.

Reply:
Yeah, I just discovered it earlier today when I was implementing my own stuff 😅  I still think you can be about as precise in that message, but it'd be something like:
```ts
new ExceedsLimitError(`constraint value for constraint ${contsraintOverLimit.contextName}`, limit)
```

We'll adjust it in that next pr